agent/configentry/merge_service_config.go

```diff
@@ -109,7 +109,7 @@ func MergeServiceConfig(defaults *structs.ServiceConfigResponse, service *struct
 	// Merge proxy defaults
 	ns := nsRaw.(*structs.NodeService)
 
-	if err := mergo.Merge(&ns.Proxy.Config, defaults.ProxyConfig); err != nil {
+	if err := mergo.Merge(&ns.Proxy.Config, defaults.ProxyConfig, mergo.WithOverride); err != nil {
```

Comment:
I'm under the impression that local service configuration is supposed to take precedence over the `service-defaults`, which would allow operators to set unique config on individual instances, if needed.

Reply:
I see; that makes sense. However, does service-defaults should precede the values from proxy-defaults? 
Values from `proxy-default` overrides `service-defaults` is also what I observed in a test.

Reply:
Typically, service-defaults should win over proxy-defaults. Are you sure that was the situation? We have lots of unit tests ensuring that service-defaults takes precedence over proxy-defaults, and I've done quite a bit of testing with it recently. If you have a specific example where it occurs, we can definitely look into it.

Reply:
I will reopen if I found any issue with the test example. Thanks.